lib/util/Components.js

```diff
@@ -114,8 +114,12 @@ Components.prototype.list = function() {
     }
     const id = this._getId(this._list[j].node);
     list[j] = this._list[j];
+    if (!list[j].usedPropTypes) {
+      list[j].usedPropTypes = [];
+    }
+
     if (usedPropTypes[id]) {
-      list[j].usedPropTypes = (list[j].usedPropTypes || []).concat(usedPropTypes[id]);
```

Comment:
this change doesn't seem like an improvement; it increases the number of mutations.